packages/cloudflare/src/cli/build/utils/extract-project-env-vars.ts

```diff
@@ -0,0 +1,38 @@
+import * as fs from "node:fs";
+import * as path from "node:path";
+
+import { parse } from "@dotenvx/dotenvx";
+import type { BuildOptions } from "@opennextjs/aws/build/helper.js";
+
+function readEnvFiles(fileNames: string[], { monorepoRoot, appPath }: BuildOptions) {
+  return fileNames
+    .flatMap((fileName) => [
+      ...(monorepoRoot !== appPath ? [path.join(monorepoRoot, fileName)] : []),
+      path.join(appPath, fileName),
+    ])
+    .filter((filePath) => fs.existsSync(filePath) && fs.statSync(filePath).isFile())
+    .map((filePath) => parse(fs.readFileSync(filePath).toString()))
+    .reduce<Record<string, string>>((acc, overrides) => ({ ...acc, ...overrides }), {});
+}
+
+/**
+ * Extracts the environment variables defined in various .env files for a project.
+ *
+ * The `NEXTJS_ENV` environment variable in `.dev.vars` determines the mode.
+ *
+ * Merged variables respect the following priority order.
+ * 1. `.env.{mode}.local`
+ * 2. `.env.local`
+ * 3. `.env.{mode}`
+ * 4. `.env`
+ *
+ * In a monorepo, the env files in an app's directory will take precedence over
+ * the env files at the root of the monorepo.
+ */
+export function extractProjectEnvVars(options: BuildOptions) {
+  const wranglerDevVars = readEnvFiles([".dev.vars"], options);
```

Comment:
What about doing this at runtime?

A wrangler "environment variable" could be defined `.dev.vars`, `wrangler.toml` and the dashboard.

We would have to embed all the `env.*` files in the code, I don't think this would be an issue?

Reply:
My initial thought was we could use the readConfig function that is on the wrangler backlog somewhere, but this makes sense.